### ISSUE TITLE
Tb 68 - when starting Thunderbird, the template submenus are intermittently not populated

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/SmartTemplates#152]
Intermittently on my Thunderbird 68 production profile, the drop down menus on the write / reply and forward menus are not populated with the file templates that are configured. The workaround is to open SmartTemplates settings (or clicking "configure menus..." from the drop down) and close them again, then the menus are populated as expected. This happens mainly if Thunderbird starts automatically when logging on to windows (Tb is automatically loaded on logon in my Windows Settings), it doesn't appear to happen when I restart Thunderbird which points to a possible race condition.

To get a proper error trace I will enable debug mode and the debug switch `extensions.smartTemplate4.debug.fileTemplates.menus`

**Comment by RealRaven2000** [comment]
I was able to reproducing this issue within a running  Windows session (by leaving Thunderbird off for a while and then clicking different folders on startup). Intersting point here is that the menus are partially populated using the new category feature: it generates empty categories but uncategorized items are still shown in the menu:

![image](https://user-images.githubusercontent.com/6314674/135636855-191e6738-4770-43ec-a966-a7e89905eef3.png)

 popup menus  are generated without any visible menu items.

Log:

```
SmartTemplates 15:24:31.446  [0 ms]	 
loadCustomMenu(false)...
SmartTemplates 15:24:31.661  [215 ms]	 
readStringFile() - Success
SmartTemplates 15:24:31.662  [1 ms]	 
promise2.then populateMenu() ...
Win error 2 during operation open on file W:\_Tb Profiles\aj1a94hh.axel\TbSync\folders68.json (The system cannot find the file specified.

) db.js:51
SmartTemplates 15:24:31.871  [209 ms]	 
readStringFile() - Success
SmartTemplates 15:24:31.871  [0 ms]	 
promise2.then populateMenu() ...
SmartTemplates 15:24:31.875  [4 ms]	 
readStringFile() - Success
SmartTemplates 15:24:31.875  [0 ms]	 
promise2.then populateMenu() ...
SmartTemplates 15:24:31.876  [1 ms]	 
readStringFile() - Success
SmartTemplates 15:24:31.876  [0 ms]	 
promise2.then populateMenu() ...
```

**Comment by RealRaven2000** [comment]
implemented in 2.17 released on 09/12/2021. 

These older versions are not accepted by addons.thunderbird.net anymore, so we are self hosting  on github:
https://github.com/RealRaven2000/SmartTemplates/releases/2.17
